packages/ui-a11y-utils/src/ScreenReaderFocusRegion.ts

```diff
@@ -115,22 +119,25 @@ class ScreenReaderFocusRegion {
 
   handleDOMMutation = (records: MutationRecord[]) => {
     records.forEach((record) => {
-      const addedNodes = Array.from(record.addedNodes) as Element[]
-      const removedNodes = Array.from(record.removedNodes) as Element[]
-
-      this.hideNodes(addedNodes)
-
-      removedNodes.forEach((removedNode) => {
-        // Node has been removed from the DOM, make sure it is
-        // removed from our list of hidden nodes as well
-        if (removedNode.tagName.toLowerCase() !== 'iframe') {
-          this.restoreNode(removedNode)
+      const addedNodes = Array.from(record.addedNodes)
+      const removedNodes = Array.from(record.removedNodes)
+
+      this.hideNodes(addedNodes.filter(isElement) as Element[])
+
+      ;(removedNodes.filter(isElement) as Element[]).forEach(
+        (removedNode: Element) => {
+          // Node has been removed from the DOM, make sure it is
+          // removed from our list of hidden nodes as well
+
+          if (removedNode.tagName.toLowerCase() !== 'iframe') {
+            this.restoreNode(removedNode as Element)
+          }
+          const iframeBodies = this.parseIframeBodies(removedNode as Element)
+          iframeBodies.forEach((iframeBody) => {
+            this.restoreNode(iframeBody)
+          })
```

Comment:
The `as Element` castings are not necessarily anymore, because the new `isElement` type guard you wrote tells TypeScript that these are indeed Elements.

```suggestion
      this.hideNodes(addedNodes.filter(isElement))

      removedNodes.filter(isElement).forEach(
        (removedNode) => {
          // Node has been removed from the DOM, make sure it is
          // removed from our list of hidden nodes as well

          if (removedNode.tagName.toLowerCase() !== 'iframe') {
            this.restoreNode(removedNode)
          }
          const iframeBodies = this.parseIframeBodies(removedNode)
          iframeBodies.forEach((iframeBody) => {
            this.restoreNode(iframeBody)
          })
```